### PR TITLE
Bringing down BuildTools into the build

### PR DESCRIPTION
**Pull request by sokket**
Adding script code to bring BuildTools down into the build as part of the init scripts

/fyi: @piotrpMSFT @eerhardt @livarcocc @brthor

### CODE REVIEW

File: init-tools.sh [hunk 1]
```diff
@@ -14,4 +14,35 @@ while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symli
 done
 DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
 
-$DIR/scripts/obtain/dotnet-install.sh --channel $CHANNEL --verbose
\ No newline at end of file
+$DIR/scripts/obtain/dotnet-install.sh --channel $CHANNEL --verbose
+
+__init_tools_log=$DIR/init-tools.log
+__PACKAGES_DIR=$DIR/.nuget/packages
+__DOTNET_PATH=$DIR/.dotnet_stage0/x64
```
Comment (eerhardt) [inline init-tools.sh]: I think this is already set. We shouldn't dupe the stage0 path here.

File: init-tools.sh [hunk 2]
```diff
@@ -14,4 +14,35 @@ while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symli
 done
 DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
 
-$DIR/scripts/obtain/dotnet-install.sh --channel $CHANNEL --verbose
\ No newline at end of file
+$DIR/scripts/obtain/dotnet-install.sh --channel $CHANNEL --verbose
+
+__init_tools_log=$DIR/init-tools.log
+__PACKAGES_DIR=$DIR/.nuget/packages
```
Comment (eerhardt) [inline init-tools.sh]: This is already set outside this script. We should dupe this path.

**Comment by eerhardt** [comment]
Just two small comments. Other than that, looks good.

:shipit:

**Comment by livarcocc** [comment]
:shipit: given that my comment is not a big deal.

**Comment by sokket** [comment]
Rebased + fixed merge issues, re-running CI to validate

**Comment by mellinoe** [comment]
Any reason we don't just take corefx's version verbatim here? There's a couple of useful things in there, like being able to override which dotnet archive is downloaded, and adding a marker file when the initialization is actually successful.

**Comment by eerhardt** [comment]
@mellinoe - a few comments on your question:

1. We don't maintain code in .cmd files - if there is a script that needs to run on Windows, we write it in .ps1.  So that's why corefx's windows script isn't taken verbatim.
2. We always use the "latest" CLI version as the "stage0", there is no reason to ever override it.  (Well, sometimes it is needed, but rarely. When we need to, we temporarily change the script to hard-code the version we need. But that gets fixed as soon as possible.)
3. Looks like the 'marker' file is added in the Windows case, but not the non-Windows case.  @sokket - In general our Windows and non-Windows scripts need to be kept in "lock step". So this is something that should be fixed in init-tools.sh. The steps it needs to do should match init-tools.ps1, and in the same order.

**Comment by sokket** [comment]
Oops sorry for leaving that out...definitely a bug. Added

**Comment by sokket** [comment]
Got two shipits! Merging. Thanks folks!

File: init-tools.sh [hunk 3]
```diff
@@ -14,4 +14,35 @@ while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symli
...
\ No newline at end of file
+$DIR/scripts/obtain/dotnet-install.sh --channel $CHANNEL --verbose
+
+__init_tools_log=$DIR/init-tools.log
+__DOTNET_CMD=$DOTNET_INSTALL_DIR/dotnet
+__BUILD_TOOLS_DIR=$DIR/build_tools
+__BUILD_TOOLS_SOURCE=https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json
+__BUILD_TOOLS_PACKAGE_VERSION=$(cat $DIR/BuildToolsVersion.txt)
+__BUILD_TOOLS_PATH=$NUGET_PACKAGES/Microsoft.DotNet.BuildTools/$__BUILD_TOOLS_PACKAGE_VERSION/lib
+__BUILD_TOOLS_SEMAPHORE=$__BUILD_TOOLS_DIR/init-tools.completed
```
Comment (eerhardt) [inline init-tools.sh]: This file doesn't ever appear to be checked.  It is just created, but no where in the script below checks for its existence.

**Pull request merged**